### PR TITLE
Add "has_issue" boolean to GET /repos/:owner/:repo API

### LINKED ISSUE #2462: has_issues boolean missing from GET /repos/:owner/:repo API
## Issue
**Impacted version**: ALL

**Deployment mode**: it doesn't matter

**Problem description**:
I was customizing [Netbeans github plugin](https://github.com/junichi11/netbeans-github-issues-plugin) to get it to work with gitbucket on HTTP when i hit a deadend: the [egit library](https://github.com/eclipse/egit-github/blob/master/org.eclipse.egit.github.core/src/org/eclipse/egit/github/core/Repository.java#L34) checks `has_issues` before listing the repo's issues as specified by [github API](https://docs.github.com/en/rest/reference/repos#get-a-repository)

### PR DESCRIPTION
**Pull request by onukura**
### Before submitting a pull-request to GitBucket I have first:

- [x] read the [contribution guidelines](https://github.com/gitbucket/gitbucket/blob/master/.github/CONTRIBUTING.md)
- [x] rebased my branch over master
- [x] verified that project is compiling
- [x] verified that tests are passing
- [x] squashed my commits as appropriate *(keep several commits if it is relevant to understand the PR)*
- [x] [marked as closed using commit message](https://help.github.com/articles/closing-issues-via-commit-messages/) all issue ID that this PR should correct

## About this PR

This PR add `has_issue` bool to response of GET /repos/:owner/:repo API and resolve #2462 .

### CODE REVIEW

File: src/main/scala/gitbucket/core/api/ApiRepository.scala [hunk 1]
```diff
@@ -39,11 +41,17 @@ object ApiRepository {
       forks = forkedCount,
       `private` = repository.isPrivate,
       default_branch = repository.defaultBranch,
-      owner = owner
+      owner = owner,
+      has_issues = issueCount > 0
```
Comment (SIkebe) [inline src/main/scala/gitbucket/core/api/ApiRepository.scala:45]: This is not the same meaning as GitHub.

image

Comment (SIkebe) [reply src/main/scala/gitbucket/core/api/ApiRepository.scala:45]: The number of issues is irrelevant here.

Comment (onukura) [reply src/main/scala/gitbucket/core/api/ApiRepository.scala:45]: @SIkebe Thanks for your review! 
Oops! I totally mistook meanings...  So it means GitBucket can have only `true` statement...?

Comment (SIkebe) [reply src/main/scala/gitbucket/core/api/ApiRepository.scala:45]: >So it means GitBucket can have only true statement...?

You can disable issue tracking here in GitBucket.

image

Comment (SIkebe) [reply src/main/scala/gitbucket/core/api/ApiRepository.scala:45]: In that case, `repository.issues_option` becomes `DISABLE`.

image

Comment (onukura) [reply src/main/scala/gitbucket/core/api/ApiRepository.scala:45]: Thank you so much for your info! that's helps a lot! 
So this API returns false when repository.issues_option is "DISABLE", got it!

Comment (SIkebe) [reply src/main/scala/gitbucket/core/api/ApiRepository.scala:45]: Yeah, I think that's the desired behavior 😉

Comment (SIkebe) [reply src/main/scala/gitbucket/core/api/ApiRepository.scala]: TBH, I wondered why GiHub returns `"has_issues": true` even if the repository has no issues at first .

Comment (onukura) [reply src/main/scala/gitbucket/core/api/ApiRepository.scala]: @SIkebe I fixed definition of `has_issue` in https://github.com/gitbucket/gitbucket/pull/2542/commits/3f33f998ca94ae0bc520ed53623618c8edd4627d

File: src/main/scala/gitbucket/core/api/ApiRepository.scala [hunk 2]
```diff
@@ -57,6 +63,7 @@ object ApiRepository {
       forks = 0,
       `private` = false,
       default_branch = "master",
-      owner = owner
+      owner = owner,
+      has_issues = false
```
Comment (SIkebe) [inline src/main/scala/gitbucket/core/api/ApiRepository.scala:67]: Super nit: Is there any reason why `has_issues` is `false` here? It's just a dummy data for web hook test, however, considering that `repository.issues.option` is default to `"PUBLIC"`, `true` might be better.

Comment (onukura) [reply src/main/scala/gitbucket/core/api/ApiRepository.scala]: yeah it is true. it is better to be `true` I fixed this too in https://github.com/gitbucket/gitbucket/pull/2542/commits/6339be6036d5ca81bad62f22929fcee61d8f99f3

**Pull request merged**